From 1b250846071221b2ff38c3ce05f5c658cd4573d6 Mon Sep 17 00:00:00 2001
From: Rashed Abdel-Tawab <rashed@linux.com>
Date: Sat, 25 Aug 2018 11:17:23 -0700
Subject: [PATCH 11/21] qcom: Declare PRODUCT_SOONG_NAMESPACES for HALs

HALs are starting to move to blueprint files, which means that pathmap
ifeqs no longer work. This results in modules with the same name in
different platform HAL directories. Since these modules aren't used or
linked anywhere outside the HAL directories and a device's device.mk,
we can safely use a soong namespace to isolate the different HAL
directories' blueprints. This defines the PRODUCT_SOONG_NAMESPACES
make flag so make knows which namespace to look in as well.

Change-Id: I549e6da863465d79b12c7453aa169d5d963afa52
---
 config/BoardConfigQcom.mk | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/config/BoardConfigQcom.mk b/config/BoardConfigQcom.mk
index cf53cf43..1569f211 100644
--- a/config/BoardConfigQcom.mk
+++ b/config/BoardConfigQcom.mk
@@ -79,6 +79,11 @@ endif
 endif
 endif
 
+PRODUCT_SOONG_NAMESPACES += \
+    hardware/qcom/audio-caf/$(QCOM_HARDWARE_VARIANT) \
+    hardware/qcom/display-caf/$(QCOM_HARDWARE_VARIANT) \
+    hardware/qcom/media-caf/$(QCOM_HARDWARE_VARIANT)
+
 # QCOM HW crypto
 ifeq ($(TARGET_HW_DISK_ENCRYPTION),true)
     TARGET_CRYPTFS_HW_PATH ?= vendor/qcom/opensource/cryptfs_hw
-- 
2.17.1

